src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -208,45 +309,85 @@ trait LazyScope
 {
   this: LazyModule =>
   override def toString: String = s"LazyScope named $name"
+  /** manage the [[LazyScope]], when calling [[apply]] function,
+    * [[LazyModule.scope]] will be altered.
+    * */
   def apply[T](body: => T) = {
-    val saved = LazyModule.scope
     LazyModule.scope = Some(this)
+    /** [[LazyModule.scope]] stack push. */
+    val saved = LazyModule.scope
+    /** evaluate [[body]].
+     * */
     val out = body
+    /** when [[body]] is evaluated, try to escape from where. */
     require (LazyModule.scope.isDefined, s"LazyScope ${name} tried to exit, but scope was empty!")
     require (LazyModule.scope.get eq this, s"LazyScope ${name} exited before LazyModule ${LazyModule.scope.get.name} was closed")
+    /** [[LazyModule.scope]] stack pop. */
     LazyModule.scope = saved
     out
   }
 }
 
+/** used to create a scope for [[LazyModule]].
+  * [[LazyModule]] can be created inside [[LazyScope]]
+  * It will instantiate a [[SimpleLazyModule]] to manage append `body`.
+  * and make `body` codes evaluated in this scope.
```

Comment:
changed to
```
/** Used to automatically create a level of module hierarchy (a [[SimpleLazyModule]]) within
  * which [[LazyModule]]s can be instantiated and connected.
  * It will instantiate a [[SimpleLazyModule]] to manage append `body`.
  * And make `body` codes evaluated in this scope.
  * */
```